src/v/pandaproxy/schema_registry/avro.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "outcome.h"
+#include "pandaproxy/schema_registry/types.h"
+
+#include <avro/ValidSchema.hh>
+
+namespace pandaproxy::schema_registry {
+
+class avro_schema_definition
+  : public named_type<avro::ValidSchema, struct avro_schema_definition_tag> {
+public:
+    using named_type<avro::ValidSchema, struct avro_schema_definition_tag>::
+      named_type;
+
+    explicit operator schema_definition() {
+        return schema_definition{_value.toJson(false)};
+    }
+
+    explicit operator schema_definition() const {
+        return schema_definition{_value.toJson(false)};
+    }
+};
```

Comment:
virtual dtor?

Reply:
> virtual dtor?

I don't think that's needed, am I missing something?

Reply:
I don't think that `avro_schema_definition` needs a virtual dtor, but it looks like named_type might? not that we are using it here in any way that would matter (eg deleting through base pointer) but this is also the first instance of named_type subclassing I've seen.

Reply:
> I don't think that `avro_schema_definition` needs a virtual dtor, but it looks like named_type might? not that we are using it here in any way that would matter (eg deleting through base pointer) but this is also the first instance of named_type subclassing I've seen.

I think the only one is `model::topic`.